File: c++/Easy/Part003/Betting game.cpp
#include <iostream>
#include <sstream>
using namespace std;

int main() {
    string diceNumbers;
    getline(cin, diceNumbers); // Reading entire line of dice numbers

    int X, Y;
    cin >> X >> Y;

    stringstream ss(diceNumbers);
    char ch;
    int number;
    int netGain = 0;

    while (ss >> ch) {
        number = ch - '0'; // Convert character to integer
        if (number % 2 == 1) {
            // Odd number
            netGain += X;
        } else {
            // Even number
            netGain -= Y;
        }
    }

    cout << netGain << endl;

    return 0;
}

File: Java/Easy/Part003/Betting game..java
import java.util.Scanner;

public class Hello {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String diceNumbers = scanner.nextLine(); // Reading entire line of dice numbers

        int X = scanner.nextInt();
        int Y = scanner.nextInt();

        int netGain = 0;

        for (char ch : diceNumbers.toCharArray()) {
            if (Character.isDigit(ch)) { // Ensure character is a digit
                int number = ch - '0'; // Convert character to integer
                if (number % 2 == 1) {
                    // Odd number
                    netGain += X;
                } else {
                    // Even number
                    netGain -= Y;
                }
            }
        }

        System.out.println(netGain);

        scanner.close();
    }
}
